include/openenclave/enclave.h

```diff
@@ -724,6 +724,61 @@ oe_enclave_t* oe_get_enclave(void);
  */
 oe_result_t oe_random(void* data, size_t size);
 
+/**
+ * oe_gen_tls_cert.
+ *
+ * This function generates a self-signed certificate with an embedded quote from
+ * the underlying enclave.
+ *
+ * @param issuer_key: a private key used to sign this certificate
+ * @param issuer_key_size The size of the issuer_key buffer.
+ * @param subject_key: a public key used to in the certificate
+ * @param subject_key_size The size of the subject_key buffer.
+ *
+ * @param output_cert: a pointer to buffer pointer, where the buffer is
+ * allocated by a client
+ * @param output_cert_size: size of the buffer above.
+ *
+ * @return OE_OK on success
+ */
+oe_result_t oe_gen_tls_cert(
+    uint8_t* issuer_key,
+    size_t issuer_key_size,
+    uint8_t* subject_key,
+    size_t subject_key_size,
+    uint8_t** output_cert,
+    size_t* output_cert_size);
+/**
+ * Free the given cert
+ * @param cert If not NULL, the buffer to free.
+ */
+void oe_free_tls_cert(uint8_t* cert);
+
+/**
+ * identity validation callback type
+ * @param identity: a pointer to an enclave's identity information
+ * @param arg: caller defined context
+ */
+typedef oe_result_t (
+    *oe_identity_verify_callback_t)(oe_identity_t* identity, void* arg);
+
+/**
+ * oe_verify_tls_cert
+ *
+ * This function preform a custom validation on the input certificate. This
+ * validation includes exacting a quote extension from the certificate before
```

Comment:
Did you mean "extracting" instead of "exacting"?

Reply:
It should be extracting. Corrected